core/services/gateway/handler_factory.go

```diff
@@ -27,10 +30,10 @@ func NewHandlerFactory(legacyChains legacyevm.LegacyChainContainer, lggr logger.
 	return &handlerFactory{legacyChains, lggr}
 }
 
-func (hf *handlerFactory) NewHandler(handlerType HandlerType, handlerConfig json.RawMessage, donConfig *config.DONConfig, don handlers.DON) (handlers.Handler, error) {
+func (hf *handlerFactory) NewHandler(handlerType HandlerType, handlerConfig json.RawMessage, donConfig *config.DONConfig, don handlers.DON, db *sqlx.DB, cfg pg.QConfig, lggr logger.Logger) (handlers.Handler, error) {
```

Comment:
Can we store these new fields in the factory struct instead of passing them to NewHandler() each time?

Reply:
thanks! looks cleaner now 👍🏼